File: codigo/05 - Association Rule Learning/1 - Apriori/apriori.R
# Apriori

# Preprocesado de Datos
library(arules)
library(arulesViz)
dataset = read.csv("Market_Basket_Optimisation.csv",header = FALSE)
dataset = read.transactions("Market_Basket_Optimisation.csv",
                            sep = ",",
                            rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset,topN=10)

# Entrenar algoritmo Apriori con el dataset
rules = apriori(data = dataset, 
                parameter = list(support = round(4*7/7500,3), confidence = 0.8/4))

# Visualización de los resultados
inspect(sort(rules, by = 'lift',decreasing = TRUE)[1:10])

# Visualizations
plot(rules, method = "graph", engine = "htmlwidget")

File: codigo/05 - Association Rule Learning/1 - Apriori/apriori.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jan  1 23:42:28 2021

@author: msantamaria
"""

# Apriori

# Cómo importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el data set 
dataset = pd.read_csv("Market_Basket_Optimisation.csv",header = None)
transactions = []
for i in range(0,7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0,20)])
    
# Entrenar el algoritmo de Apriori
from apyori import apriori
rules = apriori(transactions, 
                min_support = 0.003, 
                min_confidence = 0.2, 
                min_lift = 3, 
                min_length = 2)

# Visualización de los resultados
results = list(rules)
results[0]

the_rules = []
for result in results:
    the_rules.append({'rule': ','.join(result.items),
                      'support':result.support,
                      'confidence':result.ordered_statistics[0].confidence,
                      'lift':result.ordered_statistics[0].lift})
df = pd.DataFrame(the_rules, columns = ['rule', 'support', 'confidence', 'lift'])
